Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/EOT.cs

```diff
@@ -0,0 +1,284 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Open Rails.  If not, see <http://www.gnu.org/licenses/>.
+
+
+using Microsoft.Xna.Framework;
+using Orts.Formats.Msts;
+using Orts.Parsers.Msts;
+using Orts.Simulation.AIs;
+using Orts.Simulation.Physics;
+using Orts.Simulation.RollingStocks;
+using Orts.Simulation.RollingStocks.SubSystems;
+using Orts.Simulation.RollingStocks.SubSystems.PowerSupplies;
+using Orts.Simulation.Signalling;
+using Orts.Simulation.Timetables;
+using ORTS.Common;
+using ORTS.Scripting.Api;
+using ORTS.Settings;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Event = Orts.Common.Event;
+
+namespace Orts.Simulation.RollingStocks.SubSystems
+{
+    public enum EOTLevel
+    {
+        NoComm,
+        OneWay,
+        TwoWay
+    }
+
+    public class FullEOTPaths : List<string>
+    {
+         public FullEOTPaths (string eotPath)
+        {
+            var directories = Directory.GetDirectories(eotPath);
+            foreach (var directory in directories)
+            {
+                var files = Directory.GetFiles(directory, "*.eot");
+                foreach (var file in files)
+                {
+                    Add(file);
+                }
+            }
+        }
+    }
+    public class EOT : MSTSWagon
+    {
+        public enum EOTstate
+        {
+            Disarmed,
+            CommTestOn,
+            Armed,
+            LocalTestOn,
+            ArmNow,
+            ArmedTwoWay
+        }
+
+        public float CommTestDelayS { get; protected set; } = 5f;
+        public float LocalTestDelayS { get; protected set; } = 25f;
+
+        public static Random IDRandom = new Random();
+        public int ID;
+        public EOTstate EOTState;
+        public bool EOTEmergencyBrakingOn = false;
+        public EOTLevel EOTLevel;
+
+        protected Timer DelayTimer;
+
+        public EOT(Simulator simulator, string wagPath)
+            : base(simulator, wagPath)
+        {
+            EOTState = EOTstate.Disarmed;
+            ID = IDRandom.Next(0, 99999);
+            DelayTimer = new Timer(this);
+        }
+
+/*        public EOT(MSTSLocomotive.EOTenabled eotEnabled, bool armed, Train train)
+        {
+            Train = train;
+            EOTState = EOTstate.Disarmed;
+            EOTType = eotEnabled;
+            ID = IDRandom.Next(0, 99999);
+            if (armed)
+                EOTState = EOTstate.Armed;
+            DelayTimer = new Timer(this);
+        }*/
+
+ /*       public EOT(BinaryReader inf, Train train)
+        {
+            Train = train;
+            ID = inf.ReadInt32();
+            EOTState = (EOTstate)(inf.ReadInt32());
+            DelayTimer = new Timer(this);
+        }*/
```

Comment:
This constructor seems to be unnecessary, if the Restore() function is implemented.